src/layer/rnn.rs

```diff
@@ -73,31 +68,52 @@ impl Layer for RNN
   }
 
   fn backward(&self, params: Arc<Mutex<Params>>, delta: &Array) -> Array {
+
     // get a handle to the underlying params
     let mut ltex = params.lock().unwrap();
+
+    // update location in vector
+    ltex.current_unroll -= 1;
+
     let current_unroll = ltex.current_unroll;
-    assert!(current_unroll > 0
+    assert!(current_unroll >= 0
             , "Cannot call backward pass without at least 1 forward pass");
 
     // delta_t     = (transpose(W_{t+1}) * d_{l+1}) .* dActivation(z)
     // delta_{t-1} = (transpose(W_{t}) * d_{l})
     let dz = activations::get_derivative(&ltex.activations[0]
-                                         , &ltex.outputs[current_unroll - 1]).unwrap();
+                                         , &ltex.outputs[current_unroll]).unwrap();
     let delta_t = af::mul(delta, &dz, false);
-    let dw = af::matmul(&ltex.inputs[current_unroll - 1], &delta_t       // delta_w = delta_t * a_{t}
+    
+    let maxTime = ltex.outputs.len() - 1;
+
+    // Computation of dL/dh_{t} 
+        if current_unroll == ltex.outputs.len()-1 {
+            if ltex.deltaRec.len() == 0 {
+                ltex.deltaRec.push(delta_t.clone());
+            }
+            else {
+                ltex.deltaRec[0] = delta_t.clone();
+            }
+        }
+        else {
+            ltex.deltaRec[0] = af::add(&delta_t, &af::matmul(&ltex.deltaRec[0]
```

Comment:
So what you are saying here is that the delta from the next layer needs to be added to the delta that will be emitted to the previous layer before it can produce the gradient for the U matrix?